src/ble/link/comp_id.rs

```diff
@@ -0,0 +1,1967 @@
+//! Company identifier implementation
+
+use core::fmt;
+
+/// Company identifier for use in link layer Control PDUs
+#[derive(Copy, Clone)]
+pub struct CompanyId(u16);
+
+impl fmt::Debug for CompanyId {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "CompanyId(0x{:X})", self.as_u16())
+    }
+}
+
+impl CompanyId {
+    /// Create an unchecked company ID from a u16
+    pub fn from_raw(raw: u16) -> Self {
+        Self(raw)
+    }
+
+    /// Get inner value from company identifier
+    pub fn as_u16(&self) -> u16 {
+        self.0
+    }
+
+    /// Get the full company name from identifer
```

Comment:
Please add a warning in the docs for the binary size blowup. Something like "Warning: Using this function will add about 64 KB of string data to your application!"

Reply:
Fixed :)